extensions/native/circuit/src/jal_rangecheck/execution.rs

```diff
@@ -109,6 +109,28 @@ where
             Ok(execute_range_check_e1_impl)
         }
     }
+
+    #[cfg(feature = "tco")]
+    fn handler<Ctx: ExecutionCtxTrait>(
+        &self,
+        pc: u32,
+        inst: &Instruction<F>,
+        data: &mut [u8],
+    ) -> Result<Handler<F, Ctx>, StaticProgramError> {
+        let &Instruction { opcode, .. } = inst;
+
+        let is_jal = opcode == NativeJalOpcode::JAL.global_opcode();
+
+        if is_jal {
+            let jal_data: &mut JalPreCompute<F> = data.borrow_mut();
+            self.pre_compute_jal_impl(pc, inst, jal_data)?;
+            Ok(execute_jal_e1_tco_handler)
+        } else {
+            let range_check_data: &mut RangeCheckPreCompute = data.borrow_mut();
+            self.pre_compute_range_check_impl(pc, inst, range_check_data)?;
+            Ok(execute_range_check_e1_tco_handler)
+        }
```

Comment:
`dispatch!`?

Reply:
this one is borderline, but again feels like the code length difference is negligible